hnn_core/parallel_backends.py

```diff
@@ -260,23 +305,23 @@ def _read_stdout(self, fd, mask):
 
     def _process_child_data(self, data_bytes, data_len):
         if not data_len == len(data_bytes):
-            raise RuntimeError("Failed to receive all data from the child MPI"
-                               " process. Expecting %d bytes, got %d" %
-                               (data_len, len(data_bytes)))
+            # This is indicative of a failure. For debugging purposes.
+            warn("Length of received data unexpected. Expecting %d bytes, "
+                 "got %d" % (data_len, len(data_bytes)))
 
         if len(data_bytes) == 0:
             raise RuntimeError("MPI simulation didn't return any data")
 
         # decode base64 byte string
         try:
-            data_pickled = codecs.decode(data_bytes, "base64")
+            data_pickled = base64.b64decode(data_bytes, validate=True)
         except binascii.Error:
             # This is here for future debugging purposes. Unit tests can't
             # reproduce an incorrectly padded string, but this has been an
             # issue before
             raise ValueError("Incorrect padding for data length %d bytes" %
-                             len(data_bytes), "(mod 4 = %d)" %
-                             len(data_bytes) % 4)
+                             len(data_bytes) + " (mod 4 = %d)" %
```

Comment:
Your right that there shouldn't be a need for this, as I wouldn't expect it to hit this anymore. I think the UserWarning gives enough debugging information (actual data length and expected length) that this can be removed.

Reply:
> In the future do you think it will be necessary to add print statements indicating time passing/progress while data is transferred? As we add the ability to record from many cells simultaneously, I can see the wait times getting pretty large.

That seems like a good idea to me.

Reply:
> > In the future do you think it will be necessary to add print statements indicating time passing/progress while data is transferred? As we add the ability to record from many cells simultaneously, I can see the wait times getting pretty large.
> 
> That seems like a good idea to me.

I'm going to punt this to later if/when we see that the delay is unexpectedly long.